plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,172 @@
+    description:
+      - Override the default avatar of the webhook
+    type: str
+  tts:
+    description:
+      - True if this is a TTS message
+      - Text to speech message
+    type: bool
+  embeds:
+    description:
+      - Embedded rich content
+      - Send messages as Embeds
+    type: str
+'''
+
+EXAMPLES = """
+- name: Send a message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+
+- name: Send a message to the discord channel with specific username and avatar
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+      embeds:
+        - title: "Embedded message"
+          description: "This is an embedded message"
+          footer:
+            text: "Author: Ansible"
+          image:
+            url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str', required=False),
+            username=dict(type='str', required=False),
+            avatar_url=dict(type='str', required=False),
+            tts=dict(type='bool', required=False, default=False),
+            embeds=dict(type='list', required=False),
+        ),
+        required_one_of=[['content', 'embeds']],
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        stdout='',
+        stderr='',
+        http_code='',
+        rc=0
+    )
+
+    if module.check_mode:
+        module.exit_json(changed=False)
```

Comment:
It's not supporting `check_mode` then, right?

Reply:
I think it should return `changed=True` in check mode.

Reply:
As this module tries to send a message to a discord channel, and it's not checking whether it can connect to discord or whether credentials are fine or not, I would be confused if I'm using this module in check mode, and the module is not checking these matters. I mean this is the feeling I get for checking this module functionality.
I will not push on this, though.
In line 139 `supports_check_mode=False` should be changed to `supports_check_mode=True` as well, then.

Reply:
The problem with check mode is that everyone has different expectations what it does :-) For some people, check mode should be as close to the real thing as possible, including checking whether access tokens work etc. For other people, it suffices if `changed` is returned correctly, and not all things are checked. (I belong to the latter crowd: I use check mode to see whether a run of ansible-playbook will actually change something or not.) Since notification modules don't return sensible values for `changed` (if they can't be used to edit/update existing messages or delete existing messages, but only for sending new messages, they always return `changed == true`), one can argue whether they should support check mode at all, support it and just return `changed == true` if the parameters look sensible, or do something else.

I was assuming (and didn't check) that `supports_check_mode=True` was already there since `module.check_mode` was used. So either do not support check mode (and remove this `if`), or decide what level of check mode the module should support (and add `supports_check_mode=True`).

Reply:
`supports_check_mode=True` was already there indeed, @CWollinger changed it after my comment here: https://github.com/ansible-collections/community.general/pull/2398#discussion_r624525027

Reply:
@felixfontein by *"was already there"* you mean as "previously in the module" (as in, you're scooping for breaking changes)? If so, then it is a new module so it does not apply.

Reply:
No, in this PR, since this is a new module PR :)

Reply:
Hi all, I implemented the check mode. It is possible to send a GET request with webhook id and token. If the token and the id is correct, you will receive 200OK with some useful information in the body (e.g. channel_id, guild_id).

Reply:
Thanks for adding this feature, this would surely be very helpful for users!